.gitignore

```diff
@@ -22,3 +23,10 @@ config.mk
 
 # Debugging
 vgcore*
+*.json
+
+# Editor Misc
```

Comment:
Are these Editor misc for a particular editor? Should we keep them?

Reply:
They're popular editors and it's pretty harmless to list here, versus having them accidentally committed or everyone having to add them to `.git/info/exclude`.

Reply:
the `.vscode` folder is for VSCode. It holds miscellaneous project settings for VSCode.
`.history` is a specific [VSCode extension](https://marketplace.visualstudio.com/items?itemName=xyz.local-history) that stores copies of files whenever they're saved.

`.ccls-cache` is for the C language server, [CCLS](https://github.com/MaskRay/ccls). CCLS can be used for any editor that supports the Language Server Protocol (which in my case is both VSCode and vim). 
`.vim` is for vim, but I've only ever seen it exist when using CCLS inside vim.

As for whether to keep them, I'm not sure. I just put them there on my forked branch because I was tired of seeing 20+ files I didn't care about whenever I ran `git status`.

Reply:
Ok thanks. The reason why I was asking is that I'm revising this to fix the merge conflicts in `user-manual`

Reply:
Cool; I'd prefer to keep these ignores. Basically zero cost and makes developer life easier.

Reply:
I have always refrained to adding the QtCreator ones. I thought I was the only one using it and did not pertain other users. Can I add those too?

Reply:
Sure, go for it.

Reply:
Cool, thanks. I've updated it in the `user-manual` branch, since that file was touched for the merge conflicts anyways and the branch should be merged soon.